spec/support/shared_examples/header.rb

```diff
@@ -0,0 +1,8 @@
+RSpec.shared_examples "a page with header content" do
```

Comment:
you need to implement this in some specs or it won't be used. you may want a shared_examples for each user role to assert the correct menu is rendered for each user.

Reply:
you can also add some tests for the helper method to validate it's rendering the right links/icons etc. those don't need to be shared_examples though.